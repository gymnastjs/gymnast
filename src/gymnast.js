// @flow
import './cxs'

// Components
export { default as Col } from './col'
export { default as ConfigProvider } from './configProvider'
export { default as Grid } from './grid'
export { default as Layout } from './layout'
export { default as Offset } from './offset'
export { default as Root } from './root'

// HOC
export { default as asCol } from './asCol'
export { default as asGrid } from './asGrid'
export { default as asLayout } from './asLayout'
export { default as withResolution } from './withResolution'
export { default as withContext } from './withContext'

// Utils / Dev
export { default as Dev } from './dev'
export { default as log } from './log'
export { default as defaults } from './defaults'
export { default as utils } from './utils'

// Types
export type {
  AlignGrid,
  DisplayValues,
  Fixed,
  GridProps,
  Height,
  Justify,
  LayoutProps,
  Overflow,
  Size,
  Spacing,
  SpacingProps,
  SpacingValues,
} from './types'
