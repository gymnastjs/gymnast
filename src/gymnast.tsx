import './cxs'

// Components
export { default as Col } from './col'
export { default as GymnastProvider } from './gymnastProvider'
export { default as Grid } from './grid'

// Hooks
export { default as useGrid } from './useGrid'

// Globals
export { default as defaults } from './defaults'
export { default as log } from './log'

// Types
export {
  AlignValues,
  DisplayAliases,
  DisplayValues,
  DirectionValues,
  GridProps,
  Size,
  Spacing,
  SpacingProps,
  SpacingValues,
} from './types.d'
