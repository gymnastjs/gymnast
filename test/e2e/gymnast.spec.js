/* eslint-disable global-require, import/no-dynamic-require */
import * as Gymnast from 'gymnast'

function lowercaseFirstLetter(str = '') {
  return str[0].toLowerCase() + str.slice(1)
}

describe('e2e tests', () => {
  it('should export the same values from the main export than individual files', () => {
    expect(Gymnast).toMatchSnapshot()
  })

  it('should have all exports in predictable paths', () => {
    Object.keys(Gymnast).forEach(exportedProperty => {
      const filename = lowercaseFirstLetter(exportedProperty)
      const path = `gymnast/src/${filename}`
      const exportedValue = require(path).default

      expect(exportedValue).toBe(Gymnast[exportedProperty])
    })
  })
})
