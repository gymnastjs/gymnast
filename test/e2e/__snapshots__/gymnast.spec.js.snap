// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`e2e tests should export the same values from the main export than individual files 1`] = `
Object {
  "Col": [Function],
  "ConfigProvider": [Function],
  "Dev": [Function],
  "Grid": [Function],
  "Layout": [Function],
  "Offset": [Function],
  "Root": [Function],
  "asCol": [Function],
  "asGrid": [Function],
  "asLayout": [Function],
  "defaults": Object {
    "base": 8,
    "columns": 12,
    "displayAliases": Object {
      "large": Array [
        Object {
          "minWidth": "900px",
        },
        Object {
          "minDeviceWidth": "900px",
        },
      ],
      "medium": Array [
        Object {
          "maxWidth": "899px",
          "minWidth": "600px",
        },
        Object {
          "maxDeviceWidth": "899px",
          "minDeviceWidth": "600px",
        },
      ],
      "small": Array [
        Object {
          "maxWidth": "599px",
        },
        Object {
          "maxDeviceWidth": "599px",
        },
      ],
    },
    "fallbackDisplayKey": "default",
    "gutter": 3,
    "maxPageWidth": 153,
    "minPageWidth": 40,
    "pageMargin": Object {
      "large": 6,
      "medium": 6,
      "small": 1,
    },
    "spacingAliases": Object {
      "2XL": 6,
      "2XL/2": 3,
      "L": 3,
      "L/2": 1.5,
      "M": 2,
      "M/2": 1,
      "S": 1,
      "S/2": 0.5,
      "XL": 4,
      "XL/2": 2,
      "XS": 0.5,
    },
    "verticalGutter": 3,
  },
  "log": Object {
    "error": [Function],
    "info": [Function],
    "setLevel": [Function],
    "setLogger": [Function],
    "warn": [Function],
  },
  "utils": Object {
    "accumulateOver": [Function],
    "combineSpacing": [Function],
    "getCSS": [Function],
    "getValue": [Function],
    "getValues": [Function],
    "noop": [Function],
    "parseSpacing": [Function],
    "replaceSpacingAliases": [Function],
    "splitPattern": /\\(\\?:\\(\\?:\\\\s\\+\\)\\?,\\(\\?:\\\\s\\+\\)\\?\\|\\\\s\\+\\)/,
    "times": [Function],
    "toCXS": [Function],
    "validateSpacingProps": [Function],
  },
  "withContext": [Function],
  "withResolution": [Function],
}
`;
